frontend/tet/youth/browser-tests/search.testcafe.ts

```diff
@@ -0,0 +1,61 @@
+import { linkedEventsUrl } from '@frontend/te-yout/src/backend-api/backend-api';
+import { HttpRequestHook } from '@frontend/shared/browser-tests/http-utils/http-request-hook';
+import requestLogger, { filterLoggedRequests } from '@frontend/shared/browser-tests/utils/request-logger';
+import { clearDataToPrintOnFailure } from '@frontend/shared/browser-tests/utils/testcafe.utils';
+import frontpage from './pages/frontpage';
+import searchpage from './pages/searchpage';
+import { Selector } from 'testcafe';
+import postingpage from './pages/postingpage';
+import { getUrl } from '@frontend/shared/browser-tests/utils/url.utils';
+
+const url = getUrl(process.env.ADMIN_URL ?? 'https://localhost:3001');
+
+fixture('Frontpage')
+  .page(url)
+  .requestHooks(requestLogger, new HttpRequestHook(url, linkedEventsUrl))
+  .beforeEach(async (t) => {
+    clearDataToPrintOnFailure(t);
+  })
+  .afterEach(async () =>
+    // eslint-disable-next-line no-console
+    console.log(filterLoggedRequests(requestLogger)),
+  );
+
+test('user can search and navigate', async (t) => {
+  const currentYear: number = new Date().getFullYear();
+  const startDate = `31.12.${currentYear}`;
+  const endDate = `31.12.${currentYear + 1}`;
+  const workMethodText = 'varjosta';
```

Comment:
Itäs a bit hard to understand where these values are coming from by reading just the testing code :)
I know it's hard to write a good test against the linked events though... perhaps ultimately best test would first create the searchable item first (to linked events?) and then search it, perhaps it's something to think about later.

Reply:
It's a good question what the testing strategy should be like. Currently we aim to to test the details in unit and component tests, and the browser test is more of a proof of concept that all of the services work together. I'll add a comment clarifying this.

There is no direct need to test Linked Events here. However, there are no strong guarantees that Linked Events won't introduce changes in the future that are backwards incompatible from TET's point of view. Having an end-to-end test that highlights the corner cases how TET is using LE might be the best way to detect if a change like this has happened.